app/src/main/java/io/islnd/android/islnd/app/NotificationHelper.java

```diff
@@ -0,0 +1,172 @@
+package io.islnd.android.islnd.app;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.TaskStackBuilder;
+import android.support.v4.content.ContextCompat;
+import android.text.Spannable;
+import android.text.SpannableStringBuilder;
+import android.text.style.StyleSpan;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.islnd.android.islnd.app.activities.NavBaseActivity;
+import io.islnd.android.islnd.app.database.DataUtils;
+import io.islnd.android.islnd.app.database.IslndContract;
+import io.islnd.android.islnd.app.database.NotificationType;
+
+public class NotificationHelper {
```

Comment:
Everything is working, but I think it would be better if this listened for changes from the content provider.

We could have some type of initialization method that would register a listener to the Notification Uri. Whenever a change is reported, it would call `updateNotification`. Then `updateNotification` could be changed to private. This would help decouple the content provider from this class.

This